### PR TITLE
fix(files): add all functionality to iridium filesManager

### PR DESCRIPTION
**Pull request by josephmcg**
**What this PR does** 📖
- remove dependence on FilSystem class. previous functionality has been divided in the following ways
  - files index - iridium FilesManager store.items
  - path - files store
  - rename, like, delete - iridium FilesManager
- move ui store items to the new files module, since it's more fitting

**Which issue(s) this PR fixes** 🔨
AP-1953

**Special notes for reviewers** 🗒️
- files sidebar won't have updated values for total size. planning on implementing separately since this scope is already quite large
- this will NOT play nicely across multiple instances. 1 at a time for now. I have some plans for adding the delta later
- For now, I chose to store parentId inside each item then do a .find on a flattened file system to find the parent. The other option was circular references, but I don't believe you can do on ipfs

**Additional comments** 🎤

### CODE REVIEW

**Review by molimauro — CHANGES_REQUESTED** [review]
Super, awesome Joe. Just a minor question, thanks!!

File: components/views/files/filepath/Filepath.html [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
     <li>
-      <a @click="goBackToDirectory('root')" style="display: flex">
+      <a @click="$store.commit('files/setPath', [])" style="display: flex">
         <home-icon size="1.5x" />
       </a>
     </li>
     <li
-      v-for="name,i in path"
-      @click="goBackToDirectory(name)"
+      v-if="path.length"
```
[the remark this answers is not in the document]
Comment (josephmcg) [reply components/views/files/filepath/Filepath.html:9]: the v-if? guess not, ill remove rn

**Comment by josephmcg** [comment]
made some minor adjustments after Jason's settings manager got merged. I may adjust how the store is being handled in a separate PR. a utility he added could help simplify the issue of remote instances

File: libraries/Iridium/files/FilesManager.ts [hunk 2]
```diff
@@ -92,64 +66,121 @@ export default class FilesManager extends Emitter {
...
       size: file.size,
+      nsfw: await isNSFW(file),
+      liked: false,
+      shared: false,
+      parentId,
+      modified: Date.now(),
       type: Object.values(FILE_TYPE).includes(file.type as FILE_TYPE)
         ? (file.type as FILE_TYPE)
         : FILE_TYPE.GENERIC,
+      // TODO- maybe store thumbnail blob in ipfs as second object, then store hash here and lazy load
```
[the remark this answers is not in the document]
Comment (josephmcg) [reply libraries/Iridium/files/FilesManager.ts:132]: yep,
https://satellite-im.atlassian.net/browse/AP-1877

File: components/views/files/list/List.vue [hunk 3]
```diff
@@ -20,81 +21,42 @@ export default Vue.extend({
...
+        this.$store.commit('files/setSort', category)
+      },
+    },
   },
   mounted() {
-    this.$data.timer = setInterval(
-      this.forceRender,
-      this.$Config.chat.timestampUpdateInterval,
-    )
+    // this.$data.timer = setInterval(
```
[the remark this answers is not in the document]
Comment (josephmcg) [reply components/views/files/list/List.vue]: thanks, i commented out while removing a bunch of old logic. reimplemented a decent version. will refactor later
https://satellite-im.atlassian.net/browse/AP-1990

File: locales/en-US.js [hunk 4]
```diff
@@ -199,7 +199,7 @@ export default {
       errors: {
         no_empty: 'Item name must be a non empty string',
         leading_dot: 'Item name cannot begin with .',
-        invalid: 'Item name cannot contain invalid symbols',
+        invalid: 'Item name cannot contain invalid symbols (/:"*?<>|~#%&+{}\\)',
```
[the remark this answers is not in the document]
Comment (josephmcg) [reply locales/en-US.js:202]: it's straight from the regex used to check the name
https://github.com/Satellite-im/Core-PWA/blob/dev/config.ts#:~:text=//%20invalid%20characters%20in%20filesystem%20name

**Comment by josephmcg** [comment]
sidebar will have correct value on initial page load. it doesn't react to new additions, strangely. I'll set something up for that in a separate ticket
https://satellite-im.atlassian.net/browse/AP-1991

**Pull request merged**